test/e2e/kappcontroller/package_test.go

```diff
@@ -409,3 +410,23 @@ spec:
 		}
 	})
 }
+
+func TestPackageNotFound(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	k := e2e.Kubectl{t, env.Namespace, logger}
+	packageName := "foo.1.0.0"
+	expectedError := "stderr: 'Error from server (NotFound): package.data.packaging.carvel.dev \"foo.1.0.0\" not found"
+
+	logger.Section("Get Package", func() {
+		_, err := k.RunWithOpts([]string{"get", "package", packageName}, e2e.RunOpts{AllowError: true})
+		assert.NotNil(t, err)
+		assert.ErrorContains(t, err, expectedError)
+	})
+
+	logger.Section("delete Package", func() {
+		_, err := k.RunWithOpts([]string{"delete", "package", packageName}, e2e.RunOpts{AllowError: true})
+		assert.NotNil(t, err)
+		assert.ErrorContains(t, err, expectedError)
```

Comment:
```suggestion
		require.NotNil(t, err)
		require.ErrorContains(t, err, expectedError)
```
Also ErrorContains should already check if the error is not or not.